Windows development

shepherd007 opened the pull request:
These changes are touching code that is common for all supported systems. Actual Visual C++ project will be added in the next step.

Change log:
- added POSIX function stubs to the 'windows' folder
- added some ifdefs for signals usage
- added some ifdefs for network operations
- some tabs in the existing code was replaced with spaces for consistency

Notes:
- pthreads library is going to be downloaded directly by Visual (via Nuget mechanism) - there is no need to add stubs to the repo

Commit:
cmdline manipulation functions for Windows OS

Commit:
Port of endianes functions for Windows OS

Commit:
POSIX stub for Windows OS

Commit:
POSIX stub for Windows OS

Commit:
Visual C++ compilation

Commit:
util.c

Visual C++ support

Commit:
Visual C++ compilation

Commit:
Visual C++ development

Commit:
Visual C++ compilation

mutability commented on dump1090.c:
```diff
@@ -738,6 +744,40 @@ MODES_DUMP1090_VARIANT " " MODES_DUMP1090_VERSION
+    while (llTime >= time(NULL)) {}
+}
+#endif
+
```
Why is this needed?

shepherd007 replied:
Added to be more compliant with MalcolmRobb/dump1090 repository. Now I have mixed feelings about that - do you want me to remove displaying copyrights from dump1090 and view1090?

mutability replied:
Well:
- There's no requirement in either the BSD or GPL licenses to display this at runtime;
- It's the wrong license text anyway since this fork is GPL;
- There's nothing Windows-specific about this.

Blindly copying stuff from the MalcolmRobb fork is a bad idea.

shepherd007 replied:
Sounds reasonable - I'll remove that.

mutability commented on windows/rtl-sdr.h:
```diff
@@ -0,0 +1,366 @@
+/*
```
Don't include a copy of headers available upstream.

shepherd007 replied:
Agree. To be removed.

mutability commented:
A couple of general comments:
- Not a fan of the #ifdefs. In an ideal world the #ifdefs would be limited to including a different set of headers in one place only e.g. in dump1090.h. The biggest set seems to be in anet.c; you might want to look at completely separate implementations for POSIX vis Windows; where there are POSIX dependencies in the main code (e.g. use of close()) pull those out into a wrapper function that has alternative implementations rather than an #ifdef within the main code.
- Whitespace changes get in the way of seeing what is actually changed; please avoid changed to whitespace mingled with other changes. If you want to send through a separate PR with just whitespace changes, that's fine.

Commit:
Visual C++ review changes

Commit:
Visual C++ review changes

Commit:
Visual C++ review changes

Commit:
Visual C++ review changes

Commit:
Visual C++ review changes

Commit:
Visual C++ review changes

shepherd007 commented:
Applied requested changes:
- anet.c has windows implementation in separate file
- whitespace changes were reverted and now are not part of this commit
- removed showCopyright() artifacts from all files

shepherd007 committed:
Update cprtests.c

shepherd007 committed:
Update dump1090.c

shepherd007 committed:
Update view1090.c

shepherd007 committed:
Update net_io.c

Commit:
Removing endian.h for now

Commit:
Merge branch 'windows-development' of https://github.com/shepherd007/dump1090 into windows-development

shepherd007 commented on dump1090.c:
```diff
@@ -922,9 +928,13 @@ int main(int argc, char **argv) {
     // Set sane defaults
     modesInitConfig();
 
+#ifdef WIN32
```
@mutability I would propose replacing both windows and posix section with function call:
initSignals(&Mode.exit);

shepherd007 commented on dump1090.c:
```diff
@@ -82,6 +82,7 @@ static void log_with_timestamp(const char *format, ...)
 }
 
+#ifndef WIN32
 static void sigintHandler(int dummy) {
```
@mutability I am thinking about moving both windows and posix implementations to the separate files (i.e. signaling.c). Example of implementation would be like:
posix:
static int* exit_flag = NULL;

static void sigintHandler(int dummy) 
{
 ...
}
void initSignals(int* exit)
{
    exit_flag = exit;
     signal(SIGINT, sigintHandler);
     signal(SIGTERM, sigtermHandler);
}

windows:
static int* exit_flag = NULL;

BOOL WINAPI HandlerRoutine(DWORD dwCtrlType)
{
   ...
}

void initSignals(int* exit)
{
    exit_flag = exit;
   SetConsoleCtrlHandler(HandlerRoutine, TRUE);
}

Commit:
reduction of WIN32 defs

Commit:
reduction of WIN32 defs

Commit:
Created windows stub for sys/time.h

Commit:
compiler.h is being included using force include option

Commit:
compiler.h is being included using force include option

Commit:
compiler.h is being included using force include option

shepherd007 commented:
Cleaned utils.c and cprtest.c from WIN32 defines - it is being handled without preprocesor.

Commit:
aCreated wrapper for console cls operation

Commit:
aCreated wrapper for console cls operation

Commit:
aCreated wrapper for console cls operation

Commit:
Removed some other dependencies

Commit:
Removed some other dependencies

Commit:
Removed some other dependencies

shepherd007 commented:
- decided to split net_io.c specific for windows and posix. Posix version has removed defs for WIN32
- moved logging of messages to separate logging.c file
- moved signal handlers to separate signals.c file
- moved console related functions to separate console.c file
